tools/cli/adminTaskListCommands.go

```diff
@@ -135,9 +137,34 @@ func AdminListTaskListTasks(c *cli.Context) {
 	}
 
 	req := &persistence.GetTasksRequest{NamespaceID: namespace, TaskList: tlName, TaskType: tlType, ReadLevel: minReadLvl, MaxReadLevel: &maxReadLvl}
-	tasks, err := taskManager.GetTasks(req)
-	if err != nil {
-		ErrorAndExit("Failed to get Tasks", err)
+	paginationFunc := func(paginationToken []byte) ([]interface{}, []byte, error) {
+		response, err := taskManager.GetTasks(req)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		tasks := response.Tasks
+		if workflowID != "" {
+			all := tasks
+			tasks = tasks[:0]
+
+			for _, task := range all {
+				if task.Data.WorkflowId != workflowID {
+					continue
+				}
+				if runID != "" && task.Data.RunId != runID {
+					continue
+				}
+				tasks = append(tasks, task)
+			}
```

Comment:
Isn't this better? 
```suggestion
			filteredTasks := tasks[:0]

			for _, task := range tasks {
				if task.Data.WorkflowId != workflowID {
					continue
				}
				if runID != "" && task.Data.RunId != runID {
					continue
				}
				filteredTasks = append(filteredTasks, task)
			}
			
			tasks = filteredTasks
```
It's up to you though.

Reply:
looks good, i will test with that change

Reply:
sure, looks good